rpc/src/rpc.rs

```diff
@@ -1118,15 +1190,19 @@ impl JsonRpcRequestProcessor {
         Ok(())
     }
 
-    fn check_slot_cleaned_up<T>(
+    async fn check_slot_cleaned_up<T>(
         &self,
         result: &std::result::Result<T, BlockstoreError>,
         slot: Slot,
     ) -> Result<()> {
         let first_available_block = self
-            .blockstore
-            .get_first_available_block()
-            .unwrap_or_default();
+            .runtime
+            .spawn_blocking({
+                let blockstore = Arc::clone(&self.blockstore);
+                move || blockstore.get_first_available_block().unwrap_or_default()
```

Comment:
What's the reasoning for doing this in the bg pool? Looking at the
implementation it doesn't _look_ slow?